account_vat_on_payment/account_voucher.py

```diff
@@ -22,9 +22,13 @@ def is_vat_on_payment(self, voucher):
             for line in voucher.line_ids:
                 if line.amount:
                     valid_lines += 1
+                    # Only increase counter when invoice
+                    # not set as vat on payment
                     if (
-                        line.move_line_id and line.move_line_id.invoice and
-                        line.move_line_id.invoice.vat_on_payment
+                        line.move_line_id and
+                        ((line.move_line_id.invoice and
+                          line.move_line_id.invoice.vat_on_payment) or
+                         (not line.move_line_id.invoice))
                     ):
```

Comment:
@kenvac doesn't this set as ```vat_on_p``` every voucher that is not linked to an invoice?

Reply:
@eLBati you are right. I'll change the logic.